File: arduinobot_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
// The message type is defined in the package std_msgs, in the subfolder msg, in the file String.msg
#include <std_msgs/msg/string.hpp>

class SimpleSubscriber: public rclcpp::Node{
    private:
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
    
    public:
        SimpleSubscriber(): Node("simple_sunscriber"){
            subscription_ = create_subscription<std_msgs::msg::String>("simple_line_cpp", 10, std::bind(&SimpleSubscriber::msgCallback, this, std::placeholders::_1));
            RCLCPP_INFO(get_logger(), "Simple Subscriber Started");
        }
        void msgCallback(const std_msgs::msg::String::SharedPtr);
};

void SimpleSubscriber::msgCallback(const std_msgs::msg::String::SharedPtr msg){
    RCLCPP_INFO(get_logger(), "I heard: '%s'", msg->data.c_str());
}

int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimpleSubscriber(Node):
    def __init__(self, name):
        super().__init__(name)

        # topic
        self.subscribe_topic = "simple_line"

        # subscriber
        self.subsciber = self.create_subscription(String, self.subscribe_topic, self.msg_callback, 10)

        # Feedback
        self.get_logger().info("Simple Subscriber has started")

        
    def msg_callback(self, msg):
        self.get_logger().info("Message received: " + msg.data)

    
def main():
    rclpy.init()
    simple_subscriber = SimpleSubscriber("simple_subscriber")
    rclpy.spin(simple_subscriber)
    simple_subscriber.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
        
